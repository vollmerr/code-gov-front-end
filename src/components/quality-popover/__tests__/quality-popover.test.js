import React from 'react';

import QualityPopover from 'components/quality-popover'

import { shallow } from 'enzyme';

describe('Testing <QualityPopover /> component', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<QualityPopover/>);
  });

  it('renders 1 Autocomplete with to', () => {
    expect(wrapper.length).toEqual(1);
  });
  it('tests onClick', () => {
    wrapper.instance().onClick = jest.fn();
    wrapper.instance().forceUpdate()
    wrapper.update()
    wrapper.find('div.icon').prop('onClick')();
    expect(wrapper.instance().onClick).toHaveBeenCalled();
  });
  it('tests state change', () => {
    wrapper.setState({ activated: false });
    wrapper.instance().onClick();
    const popoverState = wrapper.state('activated');
    expect(popoverState).toEqual(true);
  })
  it('should render correct', () => {
    expect(wrapper).toMatchSnapshot();
  })
})