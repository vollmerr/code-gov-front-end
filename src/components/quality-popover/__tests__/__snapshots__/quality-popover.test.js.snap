// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <QualityPopover /> component should render correct 1`] = `
<div
  style={
    Object {
      "position": "absolute",
      "right": "0",
      "top": "-25px",
    }
  }
>
  <span
    className="data-quality-title"
  >
    Data Quality Score
  </span>
  <div
    className="icon icon-help-circled popper"
    onClick={[Function]}
  >
    <div
      className="popover desktop left"
    >
      The Data Quality Score is determined by using the information provided by Agencies in their 
      <Connect(CustomLinkComponent)
        to="/about/compliance/inventory-code"
      >
        code.json
      </Connect(CustomLinkComponent)>
       and by factors such as completeness and adherence to the 
      <Connect(CustomLinkComponent)
        to="/about/compliance/inventory-code"
      >
        metadata schema
      </Connect(CustomLinkComponent)>
      .
    </div>
    <div
      className="popover mobile left"
    >
      <div
        className="popover-title"
      >
        Data Quality Score
      </div>
      The Data Quality Score is determined by using the information provided by Agencies in their 
      <Connect(CustomLinkComponent)
        to="/about/compliance/inventory-code"
      >
        code.json
      </Connect(CustomLinkComponent)>
       and by factors such as completeness and adherence to the 
      <Connect(CustomLinkComponent)
        to="/about/compliance/inventory-code"
      >
        metadata schema
      </Connect(CustomLinkComponent)>
      .
    </div>
  </div>
</div>
`;
